jenkins/security/whitesource_container_scan.groovy

```diff
@@ -0,0 +1,202 @@
+    
+    options {
+        timeout(time: 240, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '30', numToKeepStr: '30'))
+        ansiColor('xterm')
+    }
+
+    environment {
+        PODS_ON_PRIMARY = "true"
+    }
+
+
+    parameters {
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for CORTX Cluster scripts', trim: true)
+        string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re/', description: 'Repository for CORTX Cluster scripts', trim: true)
+        string(name: 'CORTX_ALL_IMAGE', defaultValue: 'ghcr.io/seagate/cortx-all:2.0.0-latest', description: 'CORTX-ALL image', trim: true)
+        string(name: 'CORTX_SERVER_IMAGE', defaultValue: 'ghcr.io/seagate/cortx-rgw:2.0.0-latest', description: 'CORTX-SERVER image', trim: true)
+        string(name: 'CORTX_DATA_IMAGE', defaultValue: 'ghcr.io/seagate/cortx-data:2.0.0-latest', description: 'CORTX-DATA image', trim: true)
+        choice(
+            name: 'DEPLOYMENT_METHOD',
+            choices: ['standard', 'data-only'],
+            description: 'Method to deploy required CORTX service. standard method will deploy all CORTX services'
+        )
+        string(name: 'SNS_CONFIG', defaultValue: '1+0+0', description: 'sns configuration for deployment. Please select value based on disks available on nodes.', trim: true)
+        string(name: 'DIX_CONFIG', defaultValue: '1+0+0', description: 'dix configuration for deployment. Please select value based on disks available on nodes.', trim: true)
+        string(name: 'CONTROL_EXTERNAL_NODEPORT', defaultValue: '31169', description: 'Port to be used for control service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTP_NODEPORT', defaultValue: '30080', description: 'HTTP Port to be used for IO service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTPS_NODEPORT', defaultValue: '30443', description: 'HTTPS to be used for IO service.', trim: true)
+        text(defaultValue: '''hostname=<hostname>,user=<user>,pass=<password>''', description: 'VM details to be used. First node will be used as Primary node', name: 'hosts')
+        choice(
+            name: 'EXTERNAL_EXPOSURE_SERVICE',
+            choices: ['LoadBalancer', 'NodePort'],
+            description: 'K8s Service to be used to expose RGW Service to outside cluster.'
+        )
+        string(name: 'WHITESOURCE_SERVER_URL', defaultValue: ' ', description: 'WhiteSource Server URL', trim: true)
+        string(name: 'API_KEY', defaultValue: ' ', description: 'apiKey', trim: true)
+        string(name: 'USER_KEY', defaultValue: ' ', description: 'userKey', trim: true)
+        string(name: 'PRODUCT_NAME', defaultValue: ' ', description: 'productName', trim: true)
+        string(name: 'DOCKER_REGISTRY', defaultValue: ' ', description: 'docker registry', trim: true)
+        string(name: 'MAIN_POD', defaultValue: ' ', description: 'mainPod details', trim: true)
```

Comment:
Please add a default value. Do we need this value as input? Maybe we can fetch these details from the primary node.

Reply:
As Discussed Swanand want the mainPod & workerpod to be parameterized. As only the vision will be a variable hence removing mainPod and worker pod variable and keeping only one as whitesource_version